youpin-api.js

```diff
@@ -20,21 +17,17 @@ class Api {
       .configure(authentication());
 
     setInterval(() => {
-      api.refreshToken();
+      Api.refreshToken();
     }, 23 * 60 * 60000); // Refresh token every 23 hours
 
-    return new Promise((resolve, reject) => {
-      app.authenticate({
-        type: 'local',
-        'email': username,
-        'password': password
-      }).then(result => {
-        this.token = app.get('token');
-        resolve(this);
-      }).catch(error => {
-        console.log(error);
-        reject(error);
-      });
+    app.authenticate({
+      type: 'local',
+      email: this.username,
+      password: this.password,
+    }).then(() => {
+      this.token = app.get('token');
```

Comment:
I noticed `resolve(this)` is removed. I'm not a Promise expert and not sure if this would be a problem. cc: @supasate